core/src/main/java/io/onetable/client/PerTableConfigImpl.java

```diff
@@ -30,13 +30,13 @@
 
 import com.google.common.base.Preconditions;
 
-import io.onetable.hudi.HudiSourceConfig;
+import io.onetable.hudi.HudiSourceConfigImpl;
 import io.onetable.iceberg.IcebergCatalogConfig;
 import io.onetable.model.sync.SyncMode;
 
 /** Represents input configuration to the sync process. */
 @Value
```

Comment:
There is a conflict here, this class is an immutable entity class that uses lombok and implements the new interface.

Reply:
This, I don't follow. I am only just learning about lombok now and need more information. There is no compile or runtime conflict here unless there is a lack of test coverage.

Reply:
@ashvina - can you give me more information on what I should do to address this?

Reply:
Lombok's `@Value` annotation will generate some code for you. Mainly it marks all fields as `private final` and adds getters for you https://projectlombok.org/features/Value

I am not sure I see why this is a conflict though @ashvina since there are no setters on the interface

Reply:
All of the getters in this class are being generated by lombok so let's just rely on those instead of also implementing them?

Reply:
hmmm, I'd expect that I implemented them due to red lines in my IDE. I'll double check.

Reply:
done.

Reply:
There is a lombok plugin for intellij that you may need to install

Reply:
@the-other-tim-brown , the `conflict` / duplication was in the getters, they were generated by Lombok and were required by the interface as well.